bin/console lint:container - Invalid definition for service "console.command.translation_lint" with "pseudo_localization"

### Symfony version(s) affected

7.2.0

### Description

I just updated my Symfony project from Symfony 7.1 to 7.2.0.  
Now, when I run the command `bin/console lint:container`, I get this error:

`[ERROR] Invalid definition for service "console.command.translation_lint": argument 1 of
         "Symfony\Component\Translation\Command\TranslationLintCommand::__construct()" accepts
         "Symfony\Contracts\Translation\TranslatorInterface&Symfony\Component\Translation\TranslatorBagInterface",
         "Symfony\Component\Translation\PseudoLocalizationTranslator" passed.`

I tried using the `--resolve-env-vars` parameter: same error.  
I also receive the same error in both the dev and production environments.

This is my `translation.yaml` file:

```yaml
# config/packages/translation.yaml
framework:
    default_locale: '%locale%' # Read from parameters.locale
    enabled_locales: '%locales%' # Read from parameters.locales
    translator:
        default_path: '%kernel.project_dir%/translations'
        # This option is used when the translation key for the current locale wasn't found | default: the value of default_locale
        fallbacks:
            - '%locale%'
        providers:
        pseudo_localization:
            # replace characters by their accented version
            accents: false
            # wrap strings with brackets
            brackets: false
            # controls how many extra characters are added to make text longer
            expansion_factor: 1
            # maintain the original HTML tags of the translated contents
            parse_html: true
            # also translate the contents of these HTML attributes
            #localizable_html_attributes: ['title']

when@prod:
    framework:
        translator:
            pseudo_localization:
                accents: false
                brackets: false
```

If I comment out the `pseudo_localization:` block, the error goes away.

I encountered the same error with the `translation.yaml` example from [https://symfony.com/doc/current/translation.html](https://symfony.com/doc/current/translation.html).

```yaml
# config/packages/translation.yaml
framework:
    translator:
        pseudo_localization:
            # replace characters by their accented version
            accents: true
            # wrap strings with brackets
            brackets: true
            # controls how many extra characters are added to make text longer
            expansion_factor: 1.4
            # maintain the original HTML tags of the translated contents
            parse_html: true
            # also translate the contents of these HTML attributes
            localizable_html_attributes: ['title']
```

### How to reproduce

Run the command `bin/console lint:container` with:

```yaml
# config/packages/translation.yaml
framework:
    translator:
        pseudo_localization:
            # replace characters by their accented version
            accents: true
            # wrap strings with brackets
            brackets: true
            # controls how many extra characters are added to make text longer
            expansion_factor: 1.4
            # maintain the original HTML tags of the translated contents
            parse_html: true
            # also translate the contents of these HTML attributes
            localizable_html_attributes: ['title']
```

### Possible Solution

_No response_

### Additional Context

_No response_

--- a/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/TranslationLintCommandPass.php
+++ b/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/TranslationLintCommandPass.php
@@ -1,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Translation\TranslatorBagInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class TranslationLintCommandPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (!$container->hasDefinition('console.command.translation_lint') || !$container->has('translator')) {
+            return;
+        }
+
+        $translatorClass = $container->getParameterBag()->resolveValue($container->findDefinition('translator')->getClass());
+
+        if (!is_subclass_of($translatorClass, TranslatorInterface::class) || !is_subclass_of($translatorClass, TranslatorBagInterface::class)) {
+            $container->removeDefinition('console.command.translation_lint');
+        }
+    }
+}

--- a/src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php
+++ b/src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php
@@ -19,6 +19,7 @@ use Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler\ProfilerPass;
 use Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler\RemoveUnusedSessionMarshallingHandlerPass;
 use Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler\TestServiceContainerRealRefPass;
 use Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler\TestServiceContainerWeakRefPass;
+use Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler\TranslationLintCommandPass;
 use Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler\UnusedTagsPass;
 use Symfony\Bundle\FrameworkBundle\DependencyInjection\VirtualRequestStackPass;
 use Symfony\Component\Cache\Adapter\ApcuAdapter;
@@ -149,6 +150,8 @@ class FrameworkBundle extends Bundle
         $this->addCompilerPassIfExists($container, AddConstraintValidatorsPass::class);
         $this->addCompilerPassIfExists($container, AddValidatorInitializersPass::class);
         $this->addCompilerPassIfExists($container, AddConsoleCommandPass::class, PassConfig::TYPE_BEFORE_REMOVING);
+        // must be registered before the AddConsoleCommandPass
+        $container->addCompilerPass(new TranslationLintCommandPass(), PassConfig::TYPE_BEFORE_REMOVING, 10);
         // must be registered as late as possible to get access to all Twig paths registered in
         // twig.template_iterator definition
         $this->addCompilerPassIfExists($container, TranslatorPass::class, PassConfig::TYPE_BEFORE_OPTIMIZATION, -32);

--- a/src/Symfony/Component/Translation/PseudoLocalizationTranslator.php
+++ b/src/Symfony/Component/Translation/PseudoLocalizationTranslator.php
@@ -11,12 +11,13 @@
 
 namespace Symfony\Component\Translation;
 
+use Symfony\Component\Translation\Exception\LogicException;
 use Symfony\Contracts\Translation\TranslatorInterface;
 
 /**
  * This translator should only be used in a development environment.
  */
-final class PseudoLocalizationTranslator implements TranslatorInterface
+final class PseudoLocalizationTranslator implements TranslatorInterface, TranslatorBagInterface
 {
     private const EXPANSION_CHARACTER = '~';
 
@@ -113,6 +114,24 @@ final class PseudoLocalizationTranslator implements TranslatorInterface
     public function getLocale(): string
     {
         return $this->translator->getLocale();
+    }
+
+    public function getCatalogue(?string $locale = null): MessageCatalogueInterface
+    {
+        if (!$this->translator instanceof TranslatorBagInterface) {
+            throw new LogicException(\sprintf('The "%s()" method cannot be called as the wrapped translator class "%s" does not implement the "%s".', __METHOD__, $this->translator::class, TranslatorBagInterface::class));
+        }
+
+        return $this->translator->getCatalogue($locale);
+    }
+
+    public function getCatalogues(): array
+    {
+        if (!$this->translator instanceof TranslatorBagInterface) {
+            throw new LogicException(\sprintf('The "%s()" method cannot be called as the wrapped translator class "%s" does not implement the "%s".', __METHOD__, $this->translator::class, TranslatorBagInterface::class));
+        }
+
+        return $this->translator->getCatalogues();
     }
 
     private function getParts(string $originalTrans): array

